packages/http/src/validator/validators/body.ts

```diff
@@ -86,20 +88,31 @@ function deserializeAndValidate(content: IMediaTypeContent, schema: JSONSchema,
   );
 }
 
+/**
+ * Given a schema, return an equivalent schema that does not include `allOf`.
+ */
+function withoutAllOf(s: JSONSchema): JSONSchema {
+  return mergeAllOf(s, {
+    ignoreAdditionalProperties: true,
+    deep: true,
+  });
+}
+
 function memoizeSchemaNormalizer(normalizer: SchemaNormalizer): SchemaNormalizer {
   const cache = new WeakMap<JSONSchema7, O.Option<JSONSchema7>>();
   return (schema: JSONSchema7) => {
     const cached = cache.get(schema);
     if (!cached) {
-      const newSchema = normalizer(schema);
+      const newSchema = normalizer(withoutAllOf(schema));
+      // const newSchema = normalizer(schema);
       cache.set(schema, newSchema);
       return newSchema;
     }
     return cached;
   };
 }
 
-type SchemaNormalizer = (schema: JSONSchema) => O.Option<JSONSchema>;
+export type SchemaNormalizer = (schema: JSONSchema) => O.Option<JSONSchema>;
```

Comment:
I don't see a way to avoid adding an export for types, similar to what I did below with `ForUnitTesting`.  Is there a better solution to keeping this type name relatively private?